gallery/src/components/NxDrawer/NxDrawerEscExample.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useState } from 'react';
+
+import { NxDropdown, NxDrawer, NxButton, useToggle } from '@sonatype/react-shared-components';
+
+export default function NxDrawerEscExample() {
+  const [showDrawer, setShowDrawer] = useState(false);
+  const [dropdownIsOpen, onToggleDropdownCollapse] = useToggle(false);
+
+  const onClick = () => { alert('click'); };
+
+  return (
+    <>
+      <NxButton id="nx-drawer-esc-open-button" onClick={() => setShowDrawer(true)}>Open Drawer</NxButton>
+      {showDrawer && (
+        <NxDrawer id="nx-drawer-esc"
+                  onCancel={() => setShowDrawer(false)}
+                  headerTitle="An Example of NxDrawer with NxDropdown Nested Inside."
```

Comment:
Our typical pattern for a header of something like this (for instance a tile header or a modal header) is for it to be a child element.  Why the different pattern here?

Reply:
> Our typical pattern for a header of something like this (for instance a tile header or a modal header) is for it to be a child element. Why the different pattern here?

Oh we initially had it as it's own convenience component, but we thought since it is required it would be simpler for the consuming dev to just have this as a prop. But since tile and modal have a required header child element, we can follow that pattern here

Reply:
> Edit: Oh hmm this might be trickier because we have the close button and it is handled internally because of the transition thinking.

Right.  I think that could be surmounted decently easily by having the `NxDrawer.Header` component automatically render the child button as a sibling of the `children` that its given.  Hooking up the behavior of the close button might be a little harder, might require a context to pass close click events up from the header to the drawer itself.

Reply:
> > Edit: Oh hmm this might be trickier because we have the close button and it is handled internally because of the transition thinking.
> 
> Right. I think that could be surmounted decently easily by having the `NxDrawer.Header` component automatically render the child button as a sibling of the `children` that its given. Hooking up the behavior of the close button might be a little harder, might require a context to pass close click events up from the header to the drawer itself.

We moved it out here: 12488dbd5e2c39b4773d4dd2b5d09442f3d9e805